freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java

```diff
@@ -43,24 +57,41 @@ public SecurityConfig findOneByStack(Stack stack) {
                 if (!saltSecurityConfig.getSaltBootSignPrivateKey().equals(saltSecurityConfig.getSaltBootSignPrivateKeyVault())) {
                     saltSecurityConfig.setSaltBootSignPrivateKeyVault(saltSecurityConfig.getSaltBootSignPrivateKey());
                 }
-                if (!saltSecurityConfig.getSaltSignPrivateKey().equals(saltSecurityConfig.getSaltPasswordVault())) {
+                if (!saltSecurityConfig.getSaltPassword().equals(saltSecurityConfig.getSaltPasswordVault())) {
                     saltSecurityConfig.setSaltPasswordVault(saltSecurityConfig.getSaltPassword());
                 }
                 if (!saltSecurityConfig.getSaltSignPrivateKey().equals(saltSecurityConfig.getSaltSignPrivateKeyVault())) {
                     saltSecurityConfig.setSaltSignPrivateKeyVault(saltSecurityConfig.getSaltSignPrivateKey());
                 }
-                saltSecurityConfig = disabledSaltSecurityConfigRepository.save(saltSecurityConfig);
+                saltSecurityConfig = saltSecurityConfigRepository.save(saltSecurityConfig);
                 securityConfig.setSaltSecurityConfig(saltSecurityConfig);
             }
         }
         return securityConfig;
     }
 
+    public void createIfDoesntExists(Long stackId) throws TransactionExecutionException {
```

Comment:
TYPO:
'createIfDoesntExist' :)

Reply:
if you don't mind I wouldn't fix this now :)